opengrok-indexer/src/main/java/org/opengrok/indexer/history/PerforceRepository.java

```diff
@@ -61,14 +62,44 @@ public PerforceRepository() {
         ignoredFiles.add(".p4config");
     }
 
+    public static String protectPerforceFilename(String name) {
+        ArrayList<SimpleImmutableEntry<String,String>>   transmap = new ArrayList<SimpleImmutableEntry<String,String>>() {
+            {
+                /* NOTE: Must replace '%' first, or that translation will */
+                /* will affect the result of the others. */
+                add(new SimpleImmutableEntry<>("%", "%25"));
+                add(new SimpleImmutableEntry<>("#", "%23"));
+                add(new SimpleImmutableEntry<>("\\*", "%2A"));
+                add(new SimpleImmutableEntry<>("@", "%40"));
+            }
+        };
+        String t = name;   /* XXX remove? */
+        for (SimpleImmutableEntry<String,String> ent : transmap) {
+            t = t.replaceAll(ent.getKey(), ent.getValue());
+        }
+        /* Instead of iterating over a list, just force each? */
```

Comment:
not a bad idea at all. keep it simple.

Reply:
So you'd prefer I just call replaceAll four times?  Easy enough, that's why I left the comment in.  I'll use that.